src/t_hash.c

```diff
@@ -1126,7 +1128,8 @@ void hashTypeSetExDone(HashTypeSetEx *ex) {
         if (ex->c) {
             server.dirty += ex->fieldDeleted + ex->fieldUpdated;
             signalModifiedKey(ex->c, ex->db, ex->key);
-            notifyKeyspaceEvent(NOTIFY_HASH, "hexpire", ex->key, ex->db->id);
+            notifyKeyspaceEvent(NOTIFY_HASH, ex->fieldDeleted ? "hexpired" : "hexpire",
+                                ex->key, ex->db->id);
```

Comment:
I think we need to apply this line only if `ex->fieldDeleted`.

Reply:
if a field just updates the TTL, but not expired, we still need a `hexpire` notification.
am i missing something?

Reply:
One question is should we send "hexpire" before "hexpired"? I guess it is not necessary.

Reply:
i don't think so, at this time, hexpire operation doesn't occur.